core/corehttp/hostname_test.go

```diff
@@ -104,60 +105,73 @@ func TestDNSPrefix(t *testing.T) {
 }
 
 func TestKnownSubdomainDetails(t *testing.T) {
-	gwSpec := config.GatewaySpec{
-		UseSubdomains: true,
-	}
-	knownGateways := map[string]config.GatewaySpec{
-		"localhost":               gwSpec,
-		"dweb.link":               gwSpec,
-		"dweb.ipfs.pvt.k12.ma.us": gwSpec, // note the sneaky ".ipfs." ;-)
-	}
+	gwLocalhost := &config.GatewaySpec{}
+	gwDweb := &config.GatewaySpec{}
+	gwLong := &config.GatewaySpec{}
+	gwWildcard1 := &config.GatewaySpec{}
+	gwWildcard2 := &config.GatewaySpec{}
```

Comment:
I'll leave @lidel answer that one. If I made a change like that it was not on purpose.

Reply:
`knownSubdomainDetails` does not check `UseSubdomains` flag  because we perform that check later in:

https://github.com/MichaelMure/go-ipfs/blob/2ff6f1a80d6a070465e8c157a750ecf17687642b/core/corehttp/hostname.go#L149-L153

Without this, `knownSubdomainDetails` request would fallback to the generic, host-agnostic gateway at `127.0.0.1:8080/ipfs/*` and  we dont want that to happen on subdomain, as illustrated in:
https://github.com/MichaelMure/go-ipfs/blob/2ff6f1a80d6a070465e8c157a750ecf17687642b/test/sharness/t0114-gateway-subdomains.sh#L646-L652


I've updated tests to be explicit and comments in [`6b6569f` (#7319)](https://github.com/ipfs/go-ipfs/pull/7319/commits/6b6569f3e5c1cb2e229b98c26c88ffa34f0abe81)

Feel free to rename `knownSubdomainDetails` to something else.